fluentui_v2button/src/main/java/com/microsoft/fluentui/button/Utils.kt

```diff
@@ -0,0 +1,137 @@
+package com.microsoft.fluentui.button
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.interaction.InteractionSource
+import androidx.compose.foundation.interaction.collectIsFocusedAsState
+import androidx.compose.foundation.interaction.collectIsHoveredAsState
+import androidx.compose.foundation.interaction.collectIsPressedAsState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.graphics.Color
+import com.microsoft.fluentui.theme.token.StateBorderStroke
+import com.microsoft.fluentui.theme.token.StateColor
+import com.microsoft.fluentui.theme.token.controlTokens.ButtonInfo
+import com.microsoft.fluentui.theme.token.controlTokens.ButtonTokens
+import com.microsoft.fluentui.theme.token.controlTokens.FABInfo
+import com.microsoft.fluentui.theme.token.controlTokens.FABTokens
+import java.security.InvalidParameterException
+
+@Composable
+fun <S, T> backgroundColor(tokens: S, info: T, enabled: Boolean, interactionSource: InteractionSource): Color {
+    if (!(tokens is ButtonTokens && info is ButtonInfo) && !(tokens is FABTokens && info is FABInfo))
+        throw InvalidParameterException()
+
+    val fetchBackgroundColor: @Composable () -> StateColor = {
+        if (tokens is ButtonTokens)
+            (tokens as ButtonTokens).backgroundColor(info as ButtonInfo)
+        else
+            (tokens as FABTokens).backgroundColor(info as FABInfo)
+    }
+
+    if (enabled) {
```

Comment:
Lets wrap this "if" block in one function, which take colorState & interactions as input & return color.

Reply:
Done